### PR TITLE
[WIP] Bump up Cluster API dependency version to v1.5.0

### PR DESCRIPTION
**Pull request by thunderboltsid**
Changes in this PR
- Upgrade CAPI Library dependency to`v1.5.0`
- Upgrade CAPI Component manifests for e2e to `v1.5.0`
- Upgrade Kubernetes library versions to `v0.27`
- Upgrade controller-runtime version to `v0.15`
- Upgrade go version in go.mod and go install scripts to `v1.21`
- Set minimum go version in go version ensure scripts to `v1.20`
- Peg kind to `v0.19` due to lack of cgroups v2 in our CI infra machines: The conformance test still seems to fail in CI (NKE clusters running CentOS 7) due to cgroupns being private which likely means that the conformance test is using another version of kind.
- Upgrade ko version in Makefile to `v0.15.1`

**How has this been tested?**
```
$ LABEL_FILTERS="capx-feature-test"  make test-e2e-calico
...
Ran 14 of 31 Specs in 3286.188 seconds
SUCCESS! -- 14 Passed | 0 Failed | 0 Pending | 17 Skipped
PASS

Ginkgo ran 1 suite in 55m8.640549975s
Test Suite Passed

$ LABEL_FILTERS="conformance"  make test-e2e-calico
...
Ran 1 of 31 Specs in 1887.614 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 30 Skipped
PASS

Ginkgo ran 1 suite in 31m41.607584374s
Test Suite Passed
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,28 +6,21 @@ require (
...
-	k8s.io/utils v0.0.0-20220823124924-e9cbc92d1a73
-	sigs.k8s.io/cluster-api v1.3.5
+	github.com/stretchr/testify v1.8.3
+	go.uber.org/zap v1.24.0
+	k8s.io/api v0.27.2
+	k8s.io/apimachinery v0.27.2
+	k8s.io/client-go v0.27.2
+	k8s.io/utils v0.0.0-20230209194617-a36077c30491
+	sigs.k8s.io/cluster-api v1.5.0
 	sigs.k8s.io/cluster-api/test v1.3.5
```
Comment (deepakm-ntnx) [inline go.mod:19]: test needs an update too to v1.5.0?

Comment (thunderboltsid) [reply go.mod:19]: A few more things need to change before this PR is ready to review.

**Comment by thunderboltsid** [comment]
Need to bump up the base image for the prow job to ensure it works with cgroups v2 https://github.com/nutanix-cloud-native/prow-config/pull/49

**Pull request closed without merging**